sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AtomXmlSerializer, HttpOperationResponse } from "@azure/core-http";
+import * as Constants from "../util/constants";
+import { serializeToAtomXmlRequest, deserializeAtomXmlResponse } from "../util/atomXmlHelper";
+
+const requestProperties = ["Name", "Filter", "Action"];
+
+export function buildRuleOptions(name: string, ruleOptions: RuleOptions = {}): InternalRuleOptions {
+  const internalRuleOptions: InternalRuleOptions = Object.assign(ruleOptions, { name: name });
+  return internalRuleOptions;
+}
+
+export function buildRule(rawRule: any): Rule | {} {
```

Comment:
Should this return `Rule | undefined` instead? I'm not sure what the value of returning an empty object is. In `buildListRulesResponse` you do a truthy check on the return value of `buildRule` and push the rule to the list if it exists. With the code as it is right now, the return value will always 'exist'.

This would mean changing `buildRuleResponse`:
```typescript
const ruleResponse: RuleResponse = Object.assign(rule, { _response: response });
```
to 
```typescript
const ruleResponse: RuleResponse = Object.assign(rule || {}, { _response: response });
```

Reply:
This one's been updated. Also ensured that the user passed in options are not mutated by Object.assign

Reply:
@ramya0820 I see @chradek's comment above still applicable. 

The comment is about the type of the return value of `buildRule()`
I agree that it should be `Rule | undefined` instead of `Rule | {}`
The way `buildRule` is used currently, the empty object will be treated as a valid success scenario.

The same comment applies to `buildTopic`, `buildQueue` and `buildSubscription`

Reply:
Updated, looks like this was missed just for Rule